src/transaction.ts

```diff
@@ -1288,6 +1325,25 @@ export class Snapshot extends EventEmitter {
     return {params, paramTypes};
   }
 
+  /**
+   * Get directed read options
+   * @private
+   * @param {google.spanner.v1.IDirectedReadOptions} directedReadOptions Request directedReadOptions object.
+   */
+  protected _getDirectedReadOptions(
+    directedReadOptions: IDirectedReadOptions | null | undefined
+  ) {
+    if (
+      !directedReadOptions &&
+      this._getSpanner().directedReadOptions &&
+      this._options.readOnly
```

Comment:
nit: double checking whether the `readOnly` handwritten property is set for partitioned reads and queries class too?

Reply:
Also single use has this property set right?
To summarize all the below interfaces should have this readOnly property set to true,
1. Single use
2. Read only
3. Partitioned reads
4. Partitioned Query

Reply:
Yes. This has been validated